rfc/external-kms.md

```diff
@@ -30,6 +30,28 @@ It's designed to support different (cloud) KMS services as backend.
 Currently, only the ClusterKMS backend is used, which uses the master secret from the Kubernetes secrets to derive the DEKs.
 The clients of the CKMS are the join-service and the CSI drivers.
 
+### Recovery
+
+When a node boots, the disk-mapper will setup the encrypted state disk during initramfs.
+To decrypt an already existing disks, the disk-mapper will ask all available joinservice instances for a decryption key for it's current disk UUID.
+In case there is no join-service instance that can provide the decryption key, manual recovery becomes necessary.
+
+To enable manual recovery the disk-mapper starts a recovery server.
+The recovery server waits for a `recover` gRPC call.
+During a `recover` call the CLI will attest the measurements of the node.
+After successful attestation the CLI will provide a disk decryption key and measurement secret for the given disk UUID.
+The measurement secret, together with a measurement salt (not secret) is used to derive the clusterID.
+
+*Changes for eKMS; regarding disk decryption:*
+* Recovery server additionally accepts one KMS URI and one storage URI instead of a masterSecret.
+* The current flow (provide masterSecret directly) will still be supported in case no eKMS is used.
```

Comment:
To make things a bit more simple I would opt for a unified systems, i.e. not keep the current recovery flow and the eKMS recovery flow separate.
This could be achieved by embedding the master secret in a cKMS storage/kms URI.

For exmaple:
Our current implementation defines the following cKMS URI `kms://cluster-kms`.
We could adapt this URI to `kms://cluster-kms?masterSecret=<masterSecret>`, which directly contains the master secret.
If we don't want to directly provide restarting nodes with the master secret, we could instead provide just the DEK used for state disk decryption.
Storage URI in this case would be the same as during init with cKMS: `storage://no-store`.
This way the proto message is independent of the used KMS backend. To recover, a node receives KMS and storage URI, uses this to set up a KMS connection (or simply extracts the key, in case of cKMS), requests a decryption key, and decrypts its state disk.